array/array.go

```diff
@@ -0,0 +1,131 @@
+package array
+
+import (
+	"errors"
+)
+
+type Array struct {
+	arr  []int
+	size int
+}
+
+func NewArray(capacity int) *Array {
+	return &Array{
+		arr: make([]int, capacity),
+	}
+}
+
+func (a *Array) Size() int {
+	return a.size
+}
+
+func (a *Array) Capacity() int {
+	return len(a.arr)
+}
+
+func (a *Array) IsEmpty() bool {
+	return a.Size() == 0
```

Comment:
Should we use `a.size` instead of `a.Size()`?

Reply:
What if I change something in the logic of computing the array size? I think it's better to change it in one function only.

Reply:
As I understand, `Size()` is a getter for `size`. If you change something in the logic of computing the array size, `size` should be changed instead of computing again in getter

Reply:
It actually is a getter in this case, but I still think there's no problem in using a getter inside the struct implementation. Just like how I use `IsEmpty()`.

Reply:
Nothing is wrong if you use `Size()` but it will take 1 more step to get the value. It's up to you.

Reply:
Oh I see! My bad. Didn't notice this.